src/beta_inc.jl

```diff
@@ -1039,6 +1042,9 @@ function _beta_inc_inv(a::Float64, b::Float64, p::Float64, q::Float64=1-p)
         x = tx
         p_approx_prev = p_approx
     end
+
+    @debug "Newton iterations didn't converge in $maxiter iterations. The result might have reduced precision."
```

Comment:
This probably deserves a higher logging level?
```suggestion
    @warn "Newton iterations didn't converge in $maxiter iterations. The result might have reduced precision."
```

Reply:
I deliberately chose `@debug` over `@warn` to avoid that we raise warnings in normal usage. These functions sometimes lose precision in corner cases and it's not something we generally warn about. Eventually, we should have a better and more granular way of providing this information. E.g. via error codes and that's why I added the `@debug` statement.